src/cli.rs

```diff
@@ -0,0 +1,733 @@
+    pub owner: Option<OwnerFilter>,
+    /// Do not descend into a different file system
+    ///
+    /// By default, fd will traverse the file system tree as far as other options
+    /// dictate. With this flag, fd ensures that it does not descend into a
+    /// different file system than the one it started in. Comparable to the -mount
+    /// or -xdev filters of find(1).
+    #[cfg(any(unix, windows))]
+    #[clap(long, aliases(&["mount", "xdev"]), hide_short_help = true)]
+    pub one_file_system: bool,
+
+    #[cfg(feature = "completions")]
+    #[clap(long, value_parser = value_parser!(Shell), hide = true, exclusive = true)]
+    gen_completions: Option<Option<Shell>>,
+
+    #[clap(flatten)]
+    _negations: Negations,
+}
+
+impl Opts {
+    pub fn search_paths(&self, default_path: &Path) -> Vec<PathBuf> {
+        // would it make sense to concatenate these?
+        let paths = if !self.path.is_empty() {
+            &self.path
+        } else if !self.search_path.is_empty() {
+            &self.search_path
+        } else {
+            return vec![self.normalize_path(default_path)];
+        };
+        paths
+            .iter()
+            .filter_map(|path| {
+                if filesystem::is_existing_directory(path) {
+                    Some(self.normalize_path(path))
+                } else {
+                    print_error(format!(
+                        "Search path '{}' is not a directory.",
+                        path.to_string_lossy()
+                    ));
+                    None
+                }
+            })
+            .collect()
+    }
+
+    fn normalize_path(&self, path: &Path) -> PathBuf {
+        if self.absolute_path {
+            filesystem::absolute_path(path.normalize().unwrap().as_path()).unwrap()
+        } else {
+            path.to_path_buf()
+        }
+    }
+
+    pub fn no_search_paths(&self) -> bool {
+        self.path.is_empty() && self.search_path.is_empty()
+    }
+
+    #[inline]
+    pub fn rg_alias_ignore(&self) -> bool {
+        self.rg_alias_hidden_ignore > 0
+    }
+
+    pub fn max_depth(&self) -> Option<usize> {
+        self.max_depth.or(self.exact_depth)
+    }
+
+    pub fn min_depth(&self) -> Option<usize> {
+        self.min_depth.or(self.exact_depth)
+    }
+
+    pub fn threads(&self) -> usize {
+        std::cmp::max(self.threads.unwrap_or_else(num_cpus::get), 1)
+    }
+
+    pub fn max_results(&self) -> Option<usize> {
+        self.max_results
+            .filter(|&m| m > 0)
+            .or_else(|| self.max_one_result.then(|| 1))
+    }
+
+    #[cfg(feature = "completions")]
+    pub fn gen_completions(&self) -> anyhow::Result<Option<Shell>> {
+        self.gen_completions
+            .map(|maybe_shell| match maybe_shell {
+                Some(sh) => Ok(sh),
+                None => guess_shell(),
+            })
+            .transpose()
+    }
+}
+
+// TODO: windows?
+#[cfg(feature = "completions")]
+fn guess_shell() -> anyhow::Result<Shell> {
```

Comment:
For windows, you could probably just assume powershell.  Most likely any alternative will be in WSL.  In very rare cases would someone actually be running cygwin/msys